test/download.spec.js

```diff
@@ -0,0 +1,128 @@
+      await page.setContent(`<a download=true href="${server.PREFIX}/download">download</a>`);
+      const [ download ] = await Promise.all([
+        page.waitForEvent('download'),
+        page.click('a')
+      ]);
+      const path = await download.path();
+      expect(fs.existsSync(path)).toBeTruthy();
+      await download.delete();
+      expect(fs.existsSync(path)).toBeFalsy();
+    });
+    it('should expose stream', async({browser, server}) => {
+      const page = await browser.newPage({ acceptDownloads: true });
+      await page.setContent(`<a download=true href="${server.PREFIX}/download">download</a>`);
+      const [ download ] = await Promise.all([
+        page.waitForEvent('download'),
+        page.click('a')
+      ]);
+      const stream = await download.createReadStream();
+      let content = '';
+      stream.on('data', data => content += data.toString());
+      await new Promise(f => stream.on('end', f));
+      expect(content).toBe('Hello world');
+      stream.close();
+    });
+    it('should delete downloads on context destruction', async({browser, server}) => {
+      const page = await browser.newPage({ acceptDownloads: true });
+      await page.setContent(`<a download=true href="${server.PREFIX}/download">download</a>`);
+      const [ download1 ] = await Promise.all([
+        page.waitForEvent('download'),
+        page.click('a')
+      ]);
+      const [ download2 ] = await Promise.all([
+        page.waitForEvent('download'),
+        page.click('a')
+      ]);
+      const path1 = await download1.path();
+      const path2 = await download2.path();
+      expect(fs.existsSync(path1)).toBeTruthy();
+      expect(fs.existsSync(path2)).toBeTruthy();
+      await page.context().close();
+      expect(fs.existsSync(path1)).toBeFalsy();
+      expect(fs.existsSync(path2)).toBeFalsy();
+    });
+    it.fail(true)('should delete downloads on browser gone', async ({ server, defaultBrowserOptions }) => {
+      const browser = await browserType.launch(defaultBrowserOptions);
+      const page = await browser.newPage({ acceptDownloads: true });
+      await page.setContent(`<a download=true href="${server.PREFIX}/download">download</a>`);
+      const [ download1 ] = await Promise.all([
+        page.waitForEvent('download'),
+        page.click('a')
+      ]);
+      const [ download2 ] = await Promise.all([
+        page.waitForEvent('download'),
+        page.click('a')
+      ]);
+      const path1 = await download1.path();
+      const path2 = await download2.path();
+      expect(fs.existsSync(path1)).toBeTruthy();
+      expect(fs.existsSync(path2)).toBeTruthy();
+      await browser.close();
+      expect(fs.existsSync(path1)).toBeFalsy();
+      expect(fs.existsSync(path2)).toBeFalsy();
+      console.log(path.join(path1, '..'));
+      expect(fs.existsSync(path.join(path1, '..'))).toBeFalsy();
+    });
+    it('make coverage happy', async({page}) => {
+      // Otherwise FF gets mad.
```

Comment:
There is a list of not covered methods in playwright.spec.js.

Reply:
Done